lib/cc/analyzer/engine_output_filter.rb

```diff
@@ -7,42 +7,26 @@ def initialize(config = {})
         @config = config
       end
 
-      def filter?(output)
-        return true unless output.present?
-
-        if (json = parse_as_json(output))
-          issue?(json) && ignore_issue?(json)
-        else
-          false
-        end
+      def filter?(issue)
+        issue?(issue) && ignore_issue?(issue)
       end
 
       private
 
-      def parse_as_json(output)
-        JSON.parse(output)
-      rescue JSON::ParserError
-        nil
+      def issue?(issue)
+        issue.type.present? && issue.type.downcase == ISSUE_TYPE
```

Comment:
I'm having some naming trouble here: hey instance of `Issue` are you an `issue`? feels wrong. I think we either

- Don't construct an `Issue` unless it's the right type, or
- Rename `Issue` to something more generic, then it makes sense to ask that thing if it's an "issue" (vs, I guess, a warning?)

WDYT?